compose/service.py

```diff
@@ -940,3 +941,20 @@ def build_ulimits(ulimit_config):
             ulimits.append(ulimit_dict)
 
     return ulimits
+
+
+def get_log_config(logging_dict):
+    log_driver = logging_dict.get('driver', "") if logging_dict else ""
+    log_options = logging_dict.get('options', None) if logging_dict else None
+    return LogConfig(
+        type=log_driver,
+        config=log_options
+    )
+
+
+def set_log_options(container_options):
+    if ('logging' in container_options):
+        logging_dict = container_options['logging']
+        container_options['log_driver'] = logging_dict.get('driver', "")
+        container_options['log_opt'] = logging_dict.get('options', None)
+        del container_options['logging']
```

Comment:
I'd prefer if we didn't perform mutation in a utility function like this, as long as there's a nice way to do it without. Maybe:

1.  Move `'logging'` out of `DOCKER_CONFIG_OPTIONS` and into `ALLOWED_KEYS`.

2.  In `in _get_container_create_options`:

    ```python
    container_options.update(self.get_logging_options())
    ```

3. `get_logging_options`:

    ```python
    def get_logging_options(self):
        logging_dict = self.options.get('logging', {})
        return {
            'log_driver': logging_dict.get('driver', ""),
            'log_opt': logging_dict.get('options', None),
        }